### PR TITLE
almo: init at 0.9.5-alpha

### PR DESCRIPTION
**Pull request by momeemt**
## Description of changes

ALMO is a markdown parser and static page generator written in C++. It builds some format into HTML files and WebAssembly binaries that include an execution and evaluation environment.

https://github.com/abap34/almo

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/al/almo/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  };
+
+  buildInputs = [
+    gcc
+    python312Packages.pybind11
+  ];
+
+  buildPhase = ''
+    make all
+  '';
```
Comment (pbsds) [inline pkgs/by-name/al/almo/package.nix:29]:
```suggestion
  makeFlags = [ "all" ];
```

or 

```suggestion
  buildPhase = ''
    runHook preBuild
    make all
    runHook postBuild
  '';
```

File: pkgs/by-name/al/almo/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  installPhase = ''
```
Comment (pbsds) [inline pkgs/by-name/al/almo/package.nix:31]: please add `runHook preInstall` and `runHook postInstall`

**Review by pbsds — COMMENTED** [review]
on x86_64-linux:

```
almo> building '/nix/store/52jzd01f3lmpvldnx2gns0ipaszn64s2-almo-v0.9.5-alpha.drv'
almo> Running phase: unpackPhase
almo> unpacking source archive /nix/store/a4zrzc6svdf6cys9pp4bs7p9965ix7in-source
almo> source root is source
almo> Running phase: patchPhase
almo> Running phase: updateAutotoolsGnuConfigScriptsPhase
almo> Running phase: configurePhase
almo> no configure script, doing nothing
almo> Running phase: buildPhase
almo> rm -rf build
almo> rm -f almo.so
almo> bash scripts/setup.sh
almo> g++ -std=c++23 build/almo.cpp -o build/almo
almo> bash scripts/pybind.sh build/pyalmo.cpp almo.so
almo> /nix/store/81xsp348yfgmaan9r5055mcdjfw7a8wc-binutils-2.42/bin/ld: cannot find dynamic_lookup: No such file or directory
almo> collect2: error: ld returned 1 exit status
almo> ビルドエラー
almo> buildPhase completed in 1 minutes 38 seconds
almo> Running phase: installPhase
almo> cp: cannot stat 'almo.so': No such file or directory
```

File: pkgs/by-name/al/almo/package.nix (hunk 1 above)
Comment (momeemt) [reply pkgs/by-name/al/almo/package.nix]: I forgot about it, thank you very much.

File: pkgs/by-name/al/almo/package.nix (hunk 2 above)
Comment (momeemt) [reply pkgs/by-name/al/almo/package.nix:29]: I added it.

**Comment by pbsds** [comment]
this fixes almo on linux:

```nix
  # remove darwin-only linker flag on linux
  postPatch = lib.optionalString (!stdenv.isDarwin) ''
    substituteInPlace scripts/pybind.sh \
      --replace-fail " -undefined dynamic_lookup" ""
  '';
```

**Comment by momeemt** [comment]
I forgot to add a new line between a title and co-author, I'll fix it.

**Comment by momeemt** [comment]
@pbsds Thanks for teaching me. I have added `postPatch` to this package.

File: 1
Comment (pbsds) [inline 1:1]: please don't add this file

Comment (momeemt) [reply 1:1]: I have removed it.

File: pkgs/by-name/al/almo/package.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
+{
+  lib,
+  fetchFromGitHub,
+  stdenv,
+  gcc,
+  python312Packages,
+}:
+let
+  version = "v0.9.5-alpha";
```
[the remark this answers is not in the document]
Comment (momeemt) [reply pkgs/by-name/al/almo/package.nix]: Thank you, I have fixed what you pointed out.

**Pull request merged**